frontend/src/features/pet/components/PetAvatar.tsx

```
@@ -0,0 +1,46 @@
+  return (
+    <div
+      className={cn(
+        'relative transition-transform duration-300',
+        animationState === 'idle' && 'animate-pet-idle',
+        animationState === 'busy' && 'animate-pet-busy',
+        animationState === 'evolving' && 'animate-pet-evolve',
+        animationState === 'gaining_exp' && 'animate-pet-gain-exp',
+        className
+      )}
+      style={{ width: size, height: size }}
+      dangerouslySetInnerHTML={{ __html: svgString }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)